tests/Data/ProgrammesDb/EntityRepository/PromotionRepository/FindPromotionsByPidTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\CoreEntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use Tests\BBC\ProgrammesPagesService\AbstractDatabaseTest;
+
+/**
+ * @coversDefaultClass BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\PromotionRepository
+ */
+class FindPromotionsByPidTest extends AbstractDatabaseTest
+{
+    private $promotionRepository;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->loadFixtures(['PromotionsFixture']);
+        $this->promotionRepository = $this->getRepository('ProgrammesPagesService:Promotion');
+    }
+
+    /**
+     * @covers ::findActivePromotionsByPid
+     */
+    public function testOnlyPromotionsAreReceived()
```

Comment:
as opposed to SupperPromotions (they are propagated and appear in descendent core_entities)